front/lib/swr.ts

```diff
@@ -252,17 +252,21 @@ export function useDocuments(
   };
 }
 
-export function useDataSources(owner: WorkspaceType) {
+export function useDataSources(
+  owner: WorkspaceType,
+  options = { readOnly: false }
+) {
+  const { readOnly } = options;
   const dataSourcesFetcher: Fetcher<GetDataSourcesResponseBody> = fetcher;
   const { data, error, mutate } = useSWR(
-    `/api/w/${owner.sId}/data_sources`,
+    readOnly ? null : `/api/w/${owner.sId}/data_sources`,
```

Comment:
recommended way by react

Reply:
The name of the parameter here should be `disabled`, and not `readOnly`. Read only is a higher level concept in here that should not be expose.
See this PR:
https://github.com/dust-tt/dust/pull/5909